pkg/resources/provisioning.cattle.io/v1/cluster/validator.go

```diff
@@ -302,10 +317,157 @@ func (p *provisioningAdmitter) validatePSACT(request *admission.Request, respons
 	return nil
 }
 
+func validateAgentDeploymentCustomization(customization *v1.AgentDeploymentCustomization, path *field.Path) field.ErrorList {
+	if customization == nil {
+		return nil
+	}
+	var errList field.ErrorList
+
+	errList = append(errList, validateAppendToleration(customization.AppendTolerations, path.Child("appendTolerations"))...)
+	errList = append(errList, validateAffinity(customization.OverrideAffinity, path.Child("overrideAffinity"))...)
+
+	return errList
+}
+func validateAffinity(overrideAffinity *k8sv1.Affinity, path *field.Path) field.ErrorList {
```

Comment:
Similiar to my another comment: can we use the function [validateAffinity](https://github.com/kubernetes/kubernetes/blob/7b1c378edd7aa6a4e48cd324ee168c54a08a7702/pkg/apis/core/validation/validation.go#L3801) from the upstream code here?